pandas/_libs/tslibs/offsets.pyx

```diff
@@ -1849,7 +1849,7 @@ cdef class BusinessDay(BusinessMixin):
         reso = get_unit_from_dtype(dtarr.dtype)
         res = self._shift_bdays(i8other, reso=reso)
         if self.offset:
-            res = res.view(dtarr.dtype) + Timedelta(self.offset)
+            res = res.view(dtarr.dtype) + Timedelta(self.offset).as_unit("s")
```

Comment:
i might be missing something, but wouldn't this regress if the offset contains sub seconds?

```python
idx = pd.date_range("2010/02/01", "2010/02/10", freq="12h", unit='ns')
off = BDay(offset=pd.Timedelta(3, unit="ns"))
t1 = idx + off
```

Reply:
dangit good catch, will update

Reply:
updated+greenish